### PR TITLE
Extract field type from filename

### LINKED ISSUE #4786: Store the field type with the field_config
Store the extension so we can rewrite it in the same format as the original

https://github.com/equinor/ert/blob/bf942a789e8f1c62127a82ab9d7f0ef87bf448fd/src/ert/_c_wrappers/enkf/enkf_main.py#L181

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #4786 

**Approach**
_Short description of the approach_

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by ManInFez** [comment]
should we store the information in the field-info.json file?

**Comment by frode-aarstad** [comment]
Jenkins test this please

**Comment by oyvindeide** [comment]
What kind of feedback does this give if the format is unrecognized?

**Comment by frode-aarstad** [comment]
In the GUI it propagates the exception up to a qt message box. In the cli I guess the exception is just dumped into the console

File: src/ert/_c_wrappers/enkf/enkf_fs.py [hunk 1]
```diff
@@ -315,6 +317,9 @@ def save_field_info(  # pylint: disable=too-many-arguments
     def field_has_info(self, key: str) -> bool:
         return self._storage.field_has_info(key)
 
+    def load_field_info(self, key: str) -> Any:
+        return self._storage.load_field_info(key)
```
Comment (dafeda) [inline src/ert/_c_wrappers/enkf/enkf_fs.py:321]: How come we don't know the return-type here?

File: tests/unit_tests/c_wrappers/integration/test_field_parameter.py [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    )
+    prop.to_file(filename, fformat=file_format)
+    return arr
+
+
+def test_unknown_file_extension(tmpdir):
+    with tmpdir.as_cwd():
+        config = dedent(
+            """
+        JOBNAME my_name%d
```
Comment (dafeda) [inline tests/unit_tests/c_wrappers/integration/test_field_parameter.py:49]: Is specifying JOBNAME neccessary here?

File: tests/unit_tests/c_wrappers/integration/test_field_parameter.py [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+        grid = xtgeo.create_box_grid(dimension=(10, 10, 1))
+        grid.to_file("MY_EGRID.EGRID", "egrid")
+
+        write_grid_property(
+            "PARAM",
+            grid,
+            "param.grdecl",
+            "grdecl",
+            (10, 10, 1),
+            np.random.random_sample(100),
```
Comment (dafeda) [inline tests/unit_tests/c_wrappers/integration/test_field_parameter.py:67]: Consider using np.random.default_rng() instead as it is recommended by numpy. Not really important here, but perhaps best to be consistent.

File: tests/unit_tests/c_wrappers/integration/test_field_parameter.py (hunk 2 above)
Comment (frode-aarstad) [reply tests/unit_tests/c_wrappers/integration/test_field_parameter.py:49]: It seems not, ill it from all tests in this file

File: src/ert/_c_wrappers/enkf/enkf_fs.py (hunk 1 above)
Comment (frode-aarstad) [reply src/ert/_c_wrappers/enkf/enkf_fs.py:320]: Guess we could always force a dict here

**Comment by frode-aarstad** [comment]
jenkins test this please

**Review by dafeda — APPROVED** [review]

**Pull request merged**